frontend/src/pages/NewStudent.vue

```diff
@@ -23,153 +24,172 @@
+          PhoneNumber: '',
           source:'',
-          selected: null,
+          EnglishProficiency: '',
           file: null,
+          Notes: '',
+          selected: {
+            NativeLanguage: ''
+          },
+          languages: [],
+          API_nativeLanguage: []
+
       }
   },
-
+  computed: {
+    // If NativeLanguage is changed, we assign it this value 
+    nativeLanguage(){
+      return this.selected ? this.selected.NativeLanguage: ''
+    },
+    
+    // Checks if required fields are empty. If required fields are empty, the Create Student Button is disabled.
+    formIsInvalid(){
+      const formFields = ["name", "PhoneNumber"].map(item => this[item])
+      if (this.selected === null) {
+        return true
+      }
+      if (formFields.includes('') || (this.selected?.NativeLanguage === '')) {
+        return true
+      }
+      return false
+    }
+  },
   watch: {
-    file: function(val){
-      this.uploadFile();
+    file (val){
+      this.uploadFile() 
     }
   },
 
-  computed: {
-    filteredLanguageDataArray() {
-        return this.languageData.filter((option) => {
-            return option
-              .toString()
-              .toLowerCase()
-              .indexOf(this.nativeLanguage.toLowerCase()) >= 0
-            })
-    },
-      filteredSourceDataArray() {
-        return this.sourceData.filter((sourceOption) => {
-            return sourceOption
-              .toString()
-              .toLowerCase()
-              .indexOf(this.source.toLowerCase()) >= 0
-            })
-    }  
+ async mounted() {
+    // Call API for Native Languages
+      await fetch("/api/nativeLanguages")
+        .then(response => response.json())
+        .then(result => this.API_nativeLanguage = result)
   },
 
-
   methods: {
     createStudent(){
-      this.$buefy.notification.open({
-        message: 'New student added. <u>View profile</u>!',
-        duration: 5000,
-        type: 'is-success',
-        position: 'is-top',
-        // color: '#57A773',
+       const studentCreate = {
+        method: "POST",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          PhoneNumber: this.PhoneNumber,
+          FullName: this.name,
+          Source: this.source,
+          NativeLanguageString: this.selected.NativeLanguage, 
+          EnglishProficiency: this.EnglishProficiency, 
+          Notes: this.Notes,
+          StatusString: "SCREENING"
+        })
+      }
+      fetch("/api/students", studentCreate)
+        .then(response => response.json()) 
+        .then(() => {
+          this.$buefy.notification.open({
+            message: 'New student added. <u>View profile</u>!',
+            duration: 5000,
+            type: 'is-success',
+            position: 'is-top',
+          })
+        })
+        .then(() => {
+          setTimeout(() => {this.$router.push({path: `/students`})} )
+        })
```

Comment:
To better handle possible failures from the backend (you should do up the error notification with the relevant other properties):

```suggestion
        .then(response => {
          if (response.status < 400) {
            this.$buefy.notification.open({
              message: 'New student added. <u>View profile</u>!',
              duration: 5000,
              type: 'is-success',
              position: 'is-top',
            })
            setTimeout(() => {this.$router.push({path: `/students`})}, 5000)
          } else {
            this.$buefy.notification.open({ message: 'Something went wrong', ... })
          }
       })
```

Reply:
Okie! Have updated the code to show different notifications based on the HTTP response status